course6/toyapp-mehrdad/displaylib/types.h

```diff
@@ -0,0 +1,47 @@
+#pragma once
+
+#include <stdint.h>
+#include <vector>
+
+namespace Display {
+
+struct Pixel
+{
+	int mX;
+	int mY;
+};
+
+struct Vector2D
+{
+	double mX;
+	double mY;
+
+	Pixel ToPixel() { return {static_cast<int>(mX), static_cast<int>(mY)}; }
+};
+
+struct Vector3D
+{
+	double mX;
+	double mY;
+	double mZ;
+};
+
+using Degree = double;
+using Radian = double;
+
+struct Size
+{
+	int mWidth;
+	int mHeight;
+};
+
+struct Color
+{
+	uint8_t mRed;
+	uint8_t mGreen;
+	uint8_t mBlue;
+
+	uint32_t GetPixelValue() const { return (uint32_t(mRed)<<16) + (uint32_t(mGreen)<<8) + mBlue; }
```

Comment:
This format is specific to the surface we're using, and IMO the conversion should be there

Reply:
Right. Will move it there.